test/replication-luatest/persistent_gc_anon_test.lua

```diff
@@ -227,3 +239,193 @@ g.test_fetch_snapshot_on_connected_replica = function(g)
         "with the same replica UUID")
     s:close()
 end
+
+g = t.group('Expiration of anonymous replicas')
+
+g.before_each(function(g)
+    g.server = server:new()
+    g.server:start()
+    g.server:exec(function()
+        box.begin()
+        box.schema.space.create('test')
+        box.space.test:create_index('pk')
+        for i = 1, 10 do
+            box.space.test:replace{i}
+        end
+        box.commit()
+    end)
+    g.s = socket_connect(g.server)
+end)
+
+g.after_each(function(g)
+    g.s:close()
+    g.server:stop()
+end)
+
+local EXPIRATION_MSG_PATTERN = "Anonymous replica %s has not been used for " ..
+    "too long so it has been deactivated"
+local EXPIRATION_FAILURE_MSG_PATTERN = "Failed to deactivate expired " ..
+    "anonymous replica %s"
+
+-- Escape UUID to pass it to regex
+local function escape_uuid(uuid)
+    return string.gsub(uuid, '%-', '%%%-')
+end
+
+g.test_expiration_fetch_snapshot = function(g)
+    g.server:exec(function()
+        box.cfg{replication_anon_gc_timeout = 0.1}
+    end)
+    local uuid = require('uuid').str()
+    write_fetch_snapshot(g.s, uuid)
+    read_fetch_snapshot_response(g)
+
+    -- Wait while the consumer will be deleted
+    g.server:exec(function()
+        t.helpers.retrying({}, function()
+            t.assert_equals(box.space._gc_consumers:select{}, {})
+        end)
+    end)
+
+    -- Expiration is logged
+    local msg = string.format(EXPIRATION_MSG_PATTERN, escape_uuid(uuid))
+    t.assert(g.server:grep_log(msg))
+end
+
+-- The case checks if the consumer is not expired when fetching snapshot
+-- for too long.
+g.test_expiration_long_fetch_snapshot = function(g)
+    t.tarantool.skip_if_not_debug()
+    local fiber = require('fiber')
+    local uuid = require('uuid').str()
+    g.server:exec(function() box.cfg{replication_anon_gc_timeout = 0.1} end)
+
+    g.server:exec(function()
```

Comment:
How about merging those 2 execs into 1? They are to the same server in a row anyway.

Reply:
Yep, somehow missed this place. Fixed!